### DISCUSSION TITLE
How can we not evaluate the background class in voc2012 dataset

### DISCUSSION THREAD

**Discussion by qiuzhixin9527** [open-mmlab/mmsegmentation#2756] [General]
Don't care about background class
![1676512920356](https://user-images.githubusercontent.com/37932939/219248510-497b93ef-8bdc-44df-b6a4-463909c9326c.jpg)

**Comment by TianWen580** [comment]
The easiest way is to just ignore it but about the class 'background' is a lot to say([ref from OpenMMLab](https://zhuanlan.zhihu.com/p/552134716)):
When your ground-truth contains the class 'background', you can set the `reduce_zero_label` to _True_ in the train pipeline, so that the background have no way to be trained and the evaluation result of background will disappeared then. Here is an example(notice the param `reduce-zero-label=True`)👇
```python
# xxxconfigs.py

...
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', reduce_zero_label=True),
    dict(type='Resize', img_scale=(2048, 512), ratio_range=(0.5, 2.0)),
    dict(type='RandomCrop', crop_size=crop_size, cat_max_ratio=0.75),
    dict(type='RandomFlip', prob=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size=crop_size, pad_val=0, seg_pad_val=255),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_semantic_seg']),
]
...
```
Meanwhile, you are required to set `CLASSES` to ('stamp', ) and to set `num_classes` to 1 so that everything is suitable.

---
最简单的方法是忽略它，但关于“背景”类别有很多可以说的([参考文章](https://zhuanlan.zhihu.com/p/552134716))：
当你的ground-truth包含“背景”类时，你可以在train pipeline中将`reduce_zero_label`设置为True，这样背景就不会参与训练，背景的评估结果也将消失。举个配置文件的栗子（注意param `reduce-zero-label=True`）👇
```python
# xxxconfigs.py

...
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', reduce_zero_label=True),
    dict(type='Resize', img_scale=(2048, 512), ratio_range=(0.5, 2.0)),
    dict(type='RandomCrop', crop_size=crop_size, cat_max_ratio=0.75),
    dict(type='RandomFlip', prob=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size=crop_size, pad_val=0, seg_pad_val=255),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_semantic_seg']),
]
...
```
同时，你需要将 `CLASSES` 设置为 ('stamp', )，并将 `num_classes` 设置为1，使参数都对得上。
